ZenPacks/zenoss/LinuxMonitor/modeler/plugins/zenoss/cmd/linux/lvm.py

```diff
@@ -155,157 +192,64 @@ def process(self, device, results, log):
-                'r': 'read-only',
-                'R': 'read-only activation of non-read-only volume'}.get(att, None)
-
-    def lv_allocation_policy(self, att):
-        # Allocation policy: (a)nywhere, (c)ontiguous, (i)nherited, c(l)ing, (n)ormal
-        return {'a': 'anywhere',
-                'c': 'contiguous',
-                'i': 'inherited',
-                'l': 'cling',
-                'n': 'normal',
-                'A': 'anywhere (Locked)',
-                'C': 'contiguous (Locked)',
-                'I': 'inherited (Locked)',
-                'L': 'cling (Locked)',
-                'N': 'normal (Locked)'}.get(att, None)
-
-    def lv_state(self, att):
-        # State: (a)ctive, (s)uspended, (I)nvalid snapshot, invalid (S)uspended snapshot,
-        # snapshot (m)erge failed, suspended snapshot (M)erge failed,
-        # mapped (d)evice present without tables, mapped device present with (i)nactive table
-        return {'a': 'active',
-                's': 'suspended',
-                'I': 'invalid snapshot',
-                'S': 'invalid suspended snapshot',
-                'm': 'snapshot merge failed',
-                'M': 'suspended snapshot merge failed',
-                'd': 'mapped device present without tables',
-                'i': 'mapped device present with inactive table'}.get(att, None)
-
-    def lv_device(self, att):
-        # device (o)pen, (X) unknown
-        return {'o': 'open',
-                'X': 'unknown'}.get(att, None)
-
-    def lv_target_type(self, att):
-        # Target  type: (C)ache, (m)irror, (r)aid, (s)napshot, (t)hin, (u)nknown, (v)irtual
-        return {'C': 'cache',
-                'm': 'mirror',
-                'r': 'raid',
-                's': 'snapshot',
-                't': 'thin',
-                'u': 'unknown',
-                'v': 'virtual'}.get(att, None)
-
-    def lv_health(self, att):
-        # Volume Health: (p)artial, (r)efresh needed, (m)ismatches exist, (w)ritemostly, (X) unknown
-        return {'p': 'partial',
-                'r': 'refresh needed',
-                'm': 'mismatches exist',
-                'w': 'writemostly',
-                'X': 'unknown'}.get(att, None)
\ No newline at end of file
+    def makeHDMap(self, columns):
+        hd_om = ObjectMap()
+        hd_om.title = columns['disk']
+        hd_om.id = 'disk-{}'.format(self.prepId(columns['disk']))
+        hd_om.size = int(columns['size'])
+        hd_om.relname = 'harddisks'
+        hd_om.modname = 'ZenPacks.zenoss.LinuxMonitor.HardDisk'
+        return hd_om
+
+    def makePVMap(self, columns):
+        # pv_name,pv_fmt,pv_attr,pv_size,pv_free,pv_uuid,vg_name
+        pv_om = ObjectMap()
+        pv_om.title = columns['pv_name']
+        pv_om.id = 'pv-{}'.format(self.prepId(columns['pv_name']))
+        pv_om.format = columns['pv_fmt']
+        pv_om.attributes = self.lvm_parser.pv_attributes(columns['pv_attr'])
+        pv_om.pvsize = int(columns['pv_size'])
```

Comment:
Setting `pv_om.pvsize` is resulting in the following warning in zenhub.log during modeling.

```
2016-02-24 19:00:47,001 INFO zen.ZenHub: Worker (307) reports 2016-02-24 19:00:47,000 WARNING zen.ApplyDataMap: getter for 'pvsize' not found on obj 'pv-dev_sda2', skipping```

This is because zenpack.yaml defines PhysicalVolume.pvsize as a *datapoint* property. This makes it a method and not something that can be modeled.